### PR TITLE
!refactor(api/share): use height in share module

### PR DESCRIPTION
**Pull request by walldiss**
Use height instead of header in share module API

### CODE REVIEW

**Review by Wondertan — COMMENTED** [review]
Swamp tests are broken as they still pass headers instead of height

File: nodebuilder/node_test.go [hunk 1]
```diff
@@ -125,34 +123,3 @@ func StartMockOtelCollectorHTTPServer(t *testing.T) (string, func()) {
 	server.EnableHTTP2 = true
 	return server.URL, server.Close
 }
-
-func TestEmptyBlockExists(t *testing.T) {
```
Comment (Wondertan) [inline nodebuilder/node_test.go:129]: That seems unrelated. Why? Even if it's not needed, let's keep PRs focused on their goal 🙏🏻

**Review by Wondertan — APPROVED** [review]

File: nodebuilder/node_test.go (hunk 1 above)
Comment (walldiss) [reply nodebuilder/node_test.go:129]: Unnecessary test, that is tricky to fix. Removal is the best solution here

Comment (Wondertan) [reply nodebuilder/node_test.go:129]: Ok, agreed

**Review by cristaloleg — APPROVED** [review]

**Comment by Wondertan** [comment]
conflicts (

**Review by renaynay — APPROVED** [review]
very quick review, thanks @walldiss

**Review by cristaloleg — APPROVED** [review]

**Comment by vgonkivs** [comment]
FYI @jcstein

**Review by vgonkivs — APPROVED** [review]

**Review by renaynay — APPROVED** [review]
